tests/integration/test_charm.py

```diff
@@ -109,11 +114,42 @@ def assert_job_status_running_success():
             job_class.Status, name=job_object.metadata.name, namespace=namespace
         ).status["conditions"][-1]["type"]
 
-        # Check wether the last status of *Job is Running/Success
+        # Check whether the last status of *Job is Running/Success
         assert job_status in [
             "Running",
             "Succeeded",
         ], f"{job_object.metadata.name} was not running or did not succeed (status == {job_status})"
 
     assert_get_job()
     assert_job_status_running_success()
+
+
+async def test_prometheus_grafana_integration(ops_test: OpsTest):
+    """Deploy prometheus, grafana and required relations, then test the metrics."""
+    prometheus = "prometheus-k8s"
+    grafana = "grafana-k8s"
+
+    await ops_test.model.deploy(prometheus, channel="latest/beta")
+    await ops_test.model.deploy(grafana, channel="latest/beta")
+    await ops_test.model.add_relation(prometheus, grafana)
+    await ops_test.model.add_relation(APP_NAME, grafana)
+    await ops_test.model.add_relation(prometheus, APP_NAME)
+
+    await ops_test.model.wait_for_idle(status="active", timeout=60 * 10)
```

Comment:
I probably want to see [this](https://github.com/canonical/envoy-operator/pull/12/files#diff-08291d7f4ad7b2caecacb6ae26af540ebbaede9fe42d5e8e1b39d2bdefe4c2d4R47) to continue with the tests.